components/datetime/src/calendar.rs

```diff
@@ -86,8 +86,8 @@ impl CldrCalendar for Ethiopic {
     }
 }
 
-pub(crate) fn potentially_fixup_calendar<C: CldrCalendar>(
-    locale: &mut DataLocale,
+pub(crate) fn check_locale<C: CldrCalendar>(
```

Comment:
Note: I may be okay just dropping this function. The only thing it prevents you from doing is "accidentally" passing a locale with a calendar keyword into a formatter for a different calendar. I don't know how useful this is, because if the developer is passing in live locale data, they will probably never test this code path, until someone comes along and causes an unexpected error for doing the right thing (setting their calendar preference). I think the type on the TypedDateTimeFormatter should just always win.

Reply:
^ @Manishearth thoughts on this?

Reply:
Hmm, that makes some sense.

Reply:
I don't have a strong opinion on this, though a log::warn may be useful

Reply:
Should I add a `log` feature to the icu_datetime crate??

Reply:
What would the developer action be when this gets logged? They'd have to strip the calendar extension from any locale they pass in. I think it's better to document that we ignore the locale's calendar, and not do any extra work trying to log something.

Reply:
https://github.com/unicode-org/icu4x/issues/2402